exchange/exchange.go

```diff
@@ -123,16 +157,57 @@ func (e *exchange) buildBidResponse(liveAdapters []openrtb_ext.BidderName, adapt
 			// Possible performance improvement by grabbing a pointer to the current seatBid element, passing it to
 			// MakeSeatBid, and then building the seatBid in place, rather than copying. Probably more confusing than
 			// its worth
-			sb := e.makeSeatBid(adapterBids[a], a, adapterExtra)
+			sb := e.makeSeatBid(adapterBids[a], a, adapterExtra, targData)
 			seatBids = append(seatBids, *sb)
 		}
 	}
+	var err1 error = nil
+	if targData.targetFlag {
+		for id, bid := range targData.bid {
+			bidExt := new(openrtb_ext.ExtBid)
+			err1 = json.Unmarshal(bid.Ext, bidExt)
+			if err1 == nil {
+				hbPbBidderKey := string(hbpbConstantKey+"_"+targData.bidder[id])
+				hbBidderBidderKey := string(hbBidderConstantKey+"_"+targData.bidder[id])
+				hbSizeBidderKey := string(hbSizeConstantKey+"_"+targData.bidder[id])
+				hbDealIdBidderKey := string(hbDealIdConstantKey+"_"+targData.bidder[id])
+				hbCacheIdBidderKey := string(hbCacheIdConstantKey+"_"+targData.bidder[id])
+				if targData.lengthMax != 0 {
```

Comment:
Not sure if I'm missing something... is this duplicate logic from `MakePrebidTargets`? It looks like you're trimming the same keys.